launcher/internal/container_controller/container_controller.go

```diff
@@ -0,0 +1,128 @@
+	"bcs.pod.launcher.intel/resources_library/utils"
+	"github.com/docker/docker/client"
+	"github.com/go-logr/logr"
+)
+
+type ContainerController interface {
+	CreateAndRunContainers(ctx context.Context, log logr.Logger) error
+	IsContainerRunning(containerID string) (bool, error)
+}
+
+type DockerContainerController struct {
+	cli *client.Client
+}
+
+func NewDockerContainerController() (*DockerContainerController, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, err
+	}
+	return &DockerContainerController{cli: cli}, nil
+}
+
+func (d *DockerContainerController) isEmptyStruct(s interface{}) bool {
+	return reflect.DeepEqual(s, reflect.Zero(reflect.TypeOf(s)).Interface())
+}
+
+// use case covers running containers on single host
+func (d *DockerContainerController) CreateAndRunContainers(ctx context.Context, launcherConfigName string, log logr.Logger) error {
+	config, err := utils.ParseLauncherConfiguration(launcherConfigName)
+	if err != nil {
+		log.Error(err, "Failed to parse launcher configuration file")
+		return err
+	}
+	if d.isEmptyStruct(config) {
+		log.Error(err, "Failed to parse launcher configuration file. Configuration is empty")
+		return err
+	}
+	if !d.isEmptyStruct(config.RunOnce.MediaProxyAgent) {
+		mcmAgentContainer := general.Containers{
+			ContainerName:   "mcm-agent", //the name is predefined because only one instance is required on the machine
+			Ip:              config.RunOnce.MediaProxyAgent.IP,
+			ExposedPort:     config.RunOnce.MediaProxyAgent.ExposedPort, //"80/tcp",
+			BindingHostPort: config.RunOnce.MediaProxyAgent.BindingHostPort,
+			Image:           config.RunOnce.MediaProxyAgent.ImageAndTag,
+		}
+		err := utils.CreateAndRunContainer(ctx, d.cli, log, mcmAgentContainer)
+		if err != nil {
+			log.Error(err, "Failed to create contianer MCM MediaProxy Agent!")
+			return err
+		}
+	} else {
+		log.Info("You did not provide information about MCM MediaProxy Agent. Omitting creation of MCM MediaProxy Agent container")
+	}
+
+	if !d.isEmptyStruct(config.RunOnce.MediaProxyMcm) {
+		mediaProxyContainer := general.Containers{
+			ContainerName:   "media-proxy", //the name is predefined because only one instance is required on the machine
+			Ip:              config.RunOnce.MediaProxyMcm.IP,
+			ExposedPort:     config.RunOnce.MediaProxyMcm.ExposedPort, //"80/tcp",
+			BindingHostPort: config.RunOnce.MediaProxyMcm.BindingHostPort,
+			Image:           config.RunOnce.MediaProxyMcm.ImageAndTag,
+			VolumeMount:     config.RunOnce.MediaProxyMcm.Volumes,
+			Privileged:      config.RunOnce.MediaProxyMcm.Privileged,
+		}
+		err := utils.CreateAndRunContainer(ctx, d.cli, log, mediaProxyContainer)
+		if err != nil {
+			log.Error(err, "Failed to create contianer MCM MediaProxy!")
+			return err
+		}
+	} else {
+		log.Info("You did not provide information about MCM MediaProxy. Omitting creation of MCM MediaProxy container")
```

Comment:
```suggestion
		log.Info("No information about MCM MediaProxy provided. Omitting creation of MCM MediaProxy container")
```

Reply:
Change the wording same way in other places where you say `You` in log messages.